integration-tests/contract-security-wrong-field-annotations.spec.ts

```diff
@@ -0,0 +1,332 @@
+
+      // Calls entrypoint: (pair %addNoAnnot nat nat))
+      // Calling Michelson code: CONTRACT %addNoAnnot (pair (nat %valueA) (nat %valueB)) ;
+      
+      try {
+      const op = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .call({
+              entrypoint: 'addNoAnnot',
+              param: {
+                valueA: 1,
+                valueB: 2,
+              },
+            })
+            .send();
+        })
+        .then((op) => {
+          return op.confirmation(3).then(() => op.hash);
+        })
+      } catch (ex) {
+        expect(ex).toEqual(expect.objectContaining({
+          message: expect.stringContaining('[a] Value is not a number: undefined')
+        }))
+      }
+    });
+
+    it('Verify call AddWrongAnnot fails due to not matching field annotations', async () => {
+      const addition = await Tezos.contract.originate({
+        code: securityWrongAnnotations,
+        init: `0`,
+      });
+      await addition.confirmation();
+      const contractAddition = await addition.contract();
+
+      // Calls entrypoint: (pair %addWrongAnnot (nat %a) (nat %b))
+      // Calling Michelson code: CONTRACT %addWrongAnnot (pair (nat %valueA) (nat %valueB)) ;
+      
+      try{
+      const op = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .call({
+              entrypoint: 'AddWrongAnnot',
+              param: {
+                a: 1,
+                b: 2,
+              },
+            })
+            .send();
+        })
+        .then((op) => {
+          return op.confirmation(3).then(() => op.hash);
+        })
+    } catch (ex) {
+      expect(ex).toEqual(expect.objectContaining({
+        message: expect.stringContaining('[valueA] Value is not a number: undefined')
+      }))
+    }
+    });
+
+    it('Verify callNoAnnot Add', async () => {
+      const addition = await Tezos.contract.originate({
+        code: securityWrongAnnotations,
+        init: `0`,
+      });
+      await addition.confirmation();
+      const contractAddition = await addition.contract();
+
+      // Calls entrypoint: (pair %add (nat %valueA) (nat %valueB))
+      // Calling Michelson code: CONTRACT %add (pair nat nat) ;
+      
+      try {
+      const op = await Tezos.contract
+        .at(contractAddition.address)
+        .then((contract) => {
+          return contract.methodsObject
+            .callNoAnnot({
+              entrypoint: 'Add',
+              param: {
+                1: 1,
+                2: 2,
+              },
+            })
+            .send();
+        })
+        .then((op) => {
+          return op.confirmation(3).then(() => op.hash);
+        })
+      } catch (ex) {
+        expect(ex).toEqual(expect.objectContaining({
+          message: expect.stringContaining('[a] Value is not a number: undefined')
```

Comment:
Same comment and for the following ones after, can you explain what is this caught error?

Reply:
I am not sure. I think maybe we might need to figure it out.